tenzir/malloc.cpp

```diff
@@ -0,0 +1,163 @@
+#include "tenzir/allocator_config.hpp"
+
+#if TENZIR_SELECT_ALLOCATOR != TENZIR_SELECT_ALLOCATOR_NONE
+
+#  include "tenzir/allocator.hpp"
+
+#  include <cerrno>
+#  include <cstddef>
+#  include <cstring>
+#  include <limits>
+#  include <new>
+
+namespace {
+
+static_assert(alignof(std::max_align_t) <= 16,
+              "We expect that mimallocs default alignment is 16 and at least "
+              "as strict as C's default alignment ");
```

Comment:
This is somewhat confusing. Mimalloc does return pointers that are aligned to 16 bytes. We refuse to build in case the platform defaults to a bigger value.